src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1208,74 +1059,169 @@ internal static unsafe bool TryParseUInt32(ReadOnlySpan<char> s, NumberStyles st
+                answer = charToHexLookup[num]; // first digit
+                index++;
+                for (int i = 0; i < 7; i++) // next 7 digits can't overflow
+                {
+                    if ((uint)index >= (uint)value.Length) goto DoneAtEnd;
+                    num = value[index];
+                    if ((uint)num >= (uint)charToHexLookup.Length || (numValue = charToHexLookup[num]) == 0xFF) goto HasTrailingChars;
+                    index++;
+                    answer = 16 * answer + numValue;
+                }
+
+                // If there's another digit, it's an overflow.
+                if ((uint)index >= (uint)value.Length) goto DoneAtEnd;
+                num = value[index];
+                if ((uint)num >= (uint)charToHexLookup.Length || (numValue = charToHexLookup[num]) == 0xFF) goto HasTrailingChars;
+                index++;
+                overflow = true;
+                if ((uint)index >= (uint)value.Length) goto DoneAtEndButPotentialOverflow;
+
+                // At this point, we're either overflowing or hitting a formatting error.
+                // Format errors take precedence for compatibility. Read through any remaining digits.
+                num = value[index];
+                while ((uint)num < (uint)charToHexLookup.Length && charToHexLookup[num] != 0xFF)
+                {
+                    index++;
+                    if ((uint)index >= (uint)value.Length) goto DoneAtEndButPotentialOverflow;
+                    num = value[index];
+                }
+                goto HasTrailingChars;
             }
-            else
+ 
+        FalseExit: // parsing failed
+            result = 0;
+            return false;
+
+        DoneAtEndButPotentialOverflow:
+            if (overflow)
             {
-                if (!NumberToUInt64(ref number, ref result))
+                failureIsOverflow = true;
+                goto FalseExit;
+            }
+        DoneAtEnd:
+            result = (uint)answer;
+            return true;
+
+        HasTrailingChars: // we've successfully parsed, but there are still remaining characters in the span
+            // Skip past trailing whitespace, then past trailing zeros, and if anything else remains, fail.
+            if (IsWhite(num))
+            {
+                if ((styles & NumberStyles.AllowTrailingWhite) == 0) goto FalseExit;
+                for (index++; index < value.Length; index++)
                 {
-                    return false;
+                    if (!IsWhite(value[index])) break;
                 }
+                if ((uint)index >= (uint)value.Length) goto DoneAtEndButPotentialOverflow;
             }
-            return true;
+
+            if (!TrailingZeros(value, index)) goto FalseExit;
+            
+            goto DoneAtEndButPotentialOverflow;
+        }
+        
+        internal static unsafe bool TryParseUInt64(ReadOnlySpan<char> value, NumberStyles styles, NumberFormatInfo info, out ulong result)
```

Comment:
Does this need to be marked as unsafe?

Reply:
Nope, will remove. In general these changes are removing a lot of unsafe code.